### ISSUE TITLE
Add NewHomes development identifier to Foundations Property Webhook

### ISSUE THREAD

**Issue by colearymyhome** [reapit/foundations#7016]
We have no way of identifying which properties that are triggering webhooks in Foundation are New Homes developments or not, and no way of being able to link development with their sub properties

Will be needed for Ed Dempsey, DNG, Sherry Fitzgerald and more

**Preliminary Specification (pending feedback)**
- Service. Reapit.Services.Properties
- Add new string property `siteType` to `PropertyModel` that exposes a string value based on the type of record in a development. Allowed values are `master` `subPlot` and `null` - the latter should be used for properties that are not part of a development and master/subPlot should be set based on the value in `prp.site`

- Mapping

| prp.Site  | siteType  |
|---|---|
| C  | subPlot  |
| M | master  |
| empty/null  | null  |

- Commentary on model property: `The type of property if part of a development site (master/subPlot)`
- We should understand if prp.Site is used for anything else, as it could be used for new block management functionality potentially so we'll need to consider and understand this

**Comment by colearymyhome** [comment]
To add, the following is an example of a master webhook we are receiving. As you can see, there is no reference to master or subplots

{
    "headers": {
        "Host": "prod-28.northeurope.logic.azure.com",
        "Reapit-Webhook-Signature": "a2xrMGpmYzM0amZoYXUwbnRqM3UyYmUxNw==",
        "X-Signature": "s:08da424d-2a13-4934-8b81-07b54752215a:1654709240:BrCaHMwpEX3G6NEaRG5sv-OkjwQgSQ6s_dGOdIQZemJkjNmAlVyvbR4dEXgHZV3xeUTUc0CO7I3N4RzF63-fBA",
        "Content-Length": "18555",
        "Content-Type": "application/json; charset=utf-8"
    },
    "body": {
        "eventId": "94054208-2b98-403d-a56c-7b7b34268a9b",
        "entityId": "RED210630",
        "customerId": "EDD",
        "eventTime": "2022-06-08T17:27:20.9872471Z",
        "topicId": "properties.modified",
        "new": {
            "id": "RED210630",
            "created": "2021-08-26T07:45:38Z",
            "modified": "2022-06-08T17:27:15Z",
            "marketingMode": "selling",
            "currency": "GBP",
            "alternateId": null,
            "address": {
                "buildingName": "",
                "buildingNumber": "1-6",
                "line1": "Bolton Grove",
                "line2": "Ballyboden Road",
                "line3": "Rathfarnham",
                "line4": "Dublin 16",
                "postcode": "",
                "countryId": "IE",
                "geolocation": {
                    "latitude": 53.284909,
                    "longitude": -6.295156
                }
            },
            "areaId": "DBL",
            "strapline": null,
            "description": "Introducing Bolton Grove, an exclusive development of 6 brand new luxury homes smartly nestled off the Ballyboden Road next to the exclusive Bolton Hall development. Bolton Grove provides a choice of homes ranging in size from 82 sq. m. to 119 sq. m. These new A rated homes have been architecturally designed to a very high specification and come with many superb features....\r\n\r\nSituated just off Ballyboden Road in the heart of Rathfarnham, Bolton Grove is within close proximity to some wonderful local amenities including St. Endas Park, Rathfarnham Village and Rathfarnham Shopping Centre to name but a few. Neighbourhood shops are available nearby on Taylors Lane while the exclusive villages of Templeogue, Terenure and Rathgar are also withing easy reach. Local primary and secondary schools are in plentiful supply and Dublin Bus provide excellent public transport facilities to the city centre and surrounding areas. The M50 motorway is also very accessible allowing easy access to the nation’s national road network. \r\n\r\n\r\n3 x 2-bedroom Ground floor units each with individual south facing gardens\r\n 2 x 3-bedroom duplex units with private balcony\r1 x 2-bedroom duplex unit with private balcony\r\r\nQuality retail facilities nearby including Dundrum Shopping Centre, Rathfarnham Shopping Centre, Rathfarnham Village and Nutgrove Retail Park\r\r\nPremier location close to St. Endas Park, Marlay Park, Bushy Park and the beautiful Dublin Mountains",
            "longDescription": "Introducing Bolton Grove, an exclusive development of 6 brand new luxury homes smartly nestled off the Ballyboden Road next to the exclusive Bolton Hall development. Bolton Grove provides a choice of homes ranging in size from 82 sq. m. to 119 sq. m. These new A rated homes have been architecturally designed to a very high specification and come with many superb features......\r\rSituated just off Ballyboden Road in the heart of Rathfarnham, Bolton Grove is within close proximity to some wonderful local amenities including St. Endas Park, Rathfarnham Village and Rathfarnham Shopping Centre to name but a few. Neighbourhood shops are available nearby on Taylors Lane while the exclusive villages of Templeogue, Terenure and Rathgar are also withing easy reach. Local primary and secondary schools are in plentiful supply and Dublin Bus provide excellent public transport facilities to the city centre and surrounding areas. The M50 motorway is also very accessible allowing easy access to the nation’s national road network.\r3 x 2-bedroom Ground floor units each with individual south facing gardens\r\r2 x 3-bedroom duplex units with private balcony\r\r1 x 2-bedroom duplex unit with private balcony\r\rQuality retail facilities nearby including Dundrum Shopping Centre, Rathfarnham Shopping\rCentre, RathfarnhamVillage and Nutgrove Retail Park\r\rPremier location close to St. Ends Park, Marlay Park, Bushy Park and the beautiful Dublin Mountains\r\rSpecification and Features:\r\rBER ‘A2’ rated energy efficient homes\r\r10 Year structural Homebond guarantee\r\rMaintenance free brick & render exterior\r\rRationel ‘Auraplus’ Aluclad windows\r\rAll windows and external doors are wired\rfor an intruder alarm\r\rCobble lock parking\r\rHigh standard of sound and heat insulation\r\rInnovative A-Class air-to-water heating system, providing energy-efficient central heating\rand hot water, at proven reduced energy cost\r\rGenerous electrical specification with recess LED lights and additionally television points\rin master bedroom\r\rWired for high-speed fibre broadband\r\rChoice of superb fitted kitchen colour with will include Quartz worktop and up stand\rand concealed under unit lighting\r\rGenerous kitchen appliance package with integrated appliances\r\rLuxurious built-in wardrobes to all bedrooms\r\rStylish bathrooms and En Suites with contemporary bathroom suites and generous tiling\r\rHeated towel rails standard to main bathroom and En Suite",
            "localAuthorityCompanyId": null,
            "summary": "• BER ‘A2’ rated energy efficient homes\r• 10 Year structural Homebond guarantee\r• Maintenance free brick & render exterior\r• Rationel ‘Auraplus’ Aluclad windows\r• All windows and external doors are wired for an intruder alarm\r• Cobble lock parking\r• High standard of sound and heat insulation\r• Innovative A-Class air-to-water heating system, providing energy-efficient central heating and hot water, at proven reduced energy cost\r• Generous electrical specification with recess LED lights and additionally television points in master bedroom\r• Wired for high-speed fibre broadband\r• Choice of superb fitted kitchen colour with will include Quartz worktop and up stand and concealed under unit lighting\r• Generous kitchen appliance package with integrated appliances\r• Luxurious built-in wardrobes to all bedrooms\r• Stylish bathrooms and En Suites with contemporary bathroom suites and generous tiling\r• Heated towel rails standard to main bathroom and En Suite",
            "departmentId": "G",
            "negotiatorId": "BMC",
            "bedrooms": 2,
            "bedroomsMax": 3,
            "receptions": 0,
            "bathrooms": 0,
            "parkingSpaces": 0,
            "councilTax": null,
            "internetAdvertising": true,
            "isExternal": false,
            "viewingArrangements": null,
            "videoUrl": null,
            "videoCaption": null,
            "video2Url": null,
            "video2Caption": null,
            "notes": "1 Bolton Grove 2 bed ground floor apt  82.3 sq. m Reserved\r\n2 Bolton Grove 2 bed duplex  91.7 sq. m. Reserved\r\n3 Bolton Grove 3 bed duplex  119.4 sq. m €585,000\r\n4 Bolton Grove 2 bed ground floor apt  84.6 sq. m. €500,000\r\n5 Bolton Grove 3 bed duplex  115.3 sq. m €570,000\r\n6 Bolton Grove 2 bed ground floor apt  84.6 sq. m. €500,000\r\n\r\n\r\n3 x 2-bedroom Ground floor units each with individual south facing gardens\r\n\r\n2 x 3-bedroom duplex units with private balcony\r\n\r\n1 x 2-bedroom duplex unit with private balcony\r\n\r\nQuality retail facilities nearby including Dundrum Shopping Centre, Rathfarnham Shopping\r\nCentre, RathfarnhamVillage and Nutgrove Retail Park\r\n\r\nPremier location close to St. Ends Park, Marlay Park, Bushy Park and the beautiful Dublin Mountains\r\n\r\nSpecification and Features:\r\n\r\nBER ‘A2’ rated energy efficient homes\r\n\r\n10 Year structural Homebond guarantee\r\n\r\nMaintenance free brick & render exterior\r\n\r\nRationel ‘Auraplus’ Aluclad windows\r\n\r\nAll windows and external doors are wired\r\nfor an intruder alarm\r\n\r\nCobble lock parking\r\n\r\nHigh standard of sound and heat insulation\r\n\r\nInnovative A-Class air-to-water heating system, providing energy-efficient central heating\r\nand hot water, at proven reduced energy cost\r\n\r\nGenerous electrical specification with recess LED lights and additionally television points\r\nin master bedroom\r\n\r\nWired for high-speed fibre broadband\r\n\r\nChoice of superb fitted kitchen colour with will include Quartz worktop and up stand\r\nand concealed under unit lighting\r\n\r\nGenerous kitchen appliance package with integrated appliances\r\n\r\nLuxurious built-in wardrobes to all bedrooms\r\n\r\nStylish bathrooms and En Suites with contemporary bathroom suites and generous tiling\r\n\r\nHeated towel rails standard to main bathroom and En Suite",
            "boardStatus": null,
            "boardNotes": null,
            "featuredImageUrl": "https://assets.reapit.net/edd/live/pictures/RED/21/RED210630_14.jpg",
            "url": "https://eddempsey.ie/properties/1-6-bolton-grove-ballyboden-road-rathfarnham-dublin-16/",
            "groundRent": 0,
            "groundRentComment": null,
            "groundRentReviewDate": null,
            "groundRentIncrease": null,
            "serviceCharge": 0,
            "serviceChargeComment": null,
            "boardUpdated": null,
            "archivedOn": null,
            "fromArchive": false,
            "externalArea": null,
            "internalArea": null,
            "epc": {
                "exempt": false,
                "eer": 0,
                "eerPotential": 0,
                "eir": 0,
                "eirPotential": 0
            },
            "selling": {
                "instructed": "2021-10-26",
                "price": 0,
                "priceTo": 0,
                "reservationFee": 0,
                "qualifier": "askingPrice",
                "status": "forSale",
                "disposal": "privateTreaty",
                "completed": null,
                "exchanged": null,
                "accountPaid": null,
                "tenure": null,
                "vendorId": "RED210630",
                "agency": "soleAgent",
                "agencyId": "SA",
                "fee": null,
                "exchangedCompanyFee": null,
                "recommendedPrice": 0,
                "brochureId": null,
                "publicBrochureUrl": null,
                "exchangedPrice": 0,
                "exchangedOfficeId": "",
                "decoration": [
                    "veryGood"
                ],
                "sharedOwnership": {
                    "sharedPercentage": null,
                    "rent": null,
                    "rentFrequency": ""
                }
            },
            "letting": null,
            "commercial": null,
            "regional": null,
            "type": [
                "flatApartment"
            ],
            "style": [
                "groundFloor",
                "upperFloor"
            ],
            "situation": [],
            "parking": [],
            "age": [
                "new"
            ],
            "locality": [],
            "specialFeatures": [],
            "rooms": [],
            "roomDetailsApproved": false,
            "officeIds": [
                "RED"
            ],
            "lostInstructionDate": null,
            "lostInstructionNote": null,
            "extrasField": {},
            "_eTag": "\"5D41CF1B4696A9BE054E47BEF4BBB745\""
        },
        "old": {
            "currency": "GBP",
            "groundRent": 0,
            "departmentId": "G",
            "internetAdvertising": true,
            "archivedOn": null,
            "longDescription": "Introducing Bolton Grove, an exclusive development of 6 brand new luxury homes smartly nestled off the Ballyboden Road next to the exclusive Bolton Hall development. Bolton Grove provides a choice of homes ranging in size from 82 sq. m. to 119 sq. m. These new A rated homes have been architecturally designed to a very high specification and come with many superb features......\r\rSituated just off Ballyboden Road in the heart of Rathfarnham, Bolton Grove is within close proximity to some wonderful local amenities including St. Endas Park, Rathfarnham Village and Rathfarnham Shopping Centre to name but a few. Neighbourhood shops are available nearby on Taylors Lane while the exclusive villages of Templeogue, Terenure and Rathgar are also withing easy reach. Local primary and secondary schools are in plentiful supply and Dublin Bus provide excellent public transport facilities to the city centre and surrounding areas. The M50 motorway is also very accessible allowing easy access to the nation’s national road network.\r3 x 2-bedroom Ground floor units each with individual south facing gardens\r\r2 x 3-bedroom duplex units with private balcony\r\r1 x 2-bedroom duplex unit with private balcony\r\rQuality retail facilities nearby including Dundrum Shopping Centre, Rathfarnham Shopping\rCentre, RathfarnhamVillage and Nutgrove Retail Park\r\rPremier location close to St. Ends Park, Marlay Park, Bushy Park and the beautiful Dublin Mountains\r\rSpecification and Features:\r\rBER ‘A2’ rated energy efficient homes\r\r10 Year structural Homebond guarantee\r\rMaintenance free brick & render exterior\r\rRationel ‘Auraplus’ Aluclad windows\r\rAll windows and external doors are wired\rfor an intruder alarm\r\rCobble lock parking\r\rHigh standard of sound and heat insulation\r\rInnovative A-Class air-to-water heating system, providing energy-efficient central heating\rand hot water, at proven reduced energy cost\r\rGenerous electrical specification with recess LED lights and additionally television points\rin master bedroom\r\rWired for high-speed fibre broadband\r\rChoice of superb fitted kitchen colour with will include Quartz worktop and up stand\rand concealed under unit lighting\r\rGenerous kitchen appliance package with integrated appliances\r\rLuxurious built-in wardrobes to all bedrooms\r\rStylish bathrooms and En Suites with contemporary bathroom suites and generous tiling\r\rHeated towel rails standard to main bathroom and En Suite",
            "parking": [],
            "extrasField": {},
            "videoUrl": null,
            "viewingArrangements": null,
            "boardNotes": null,
            "notes": "1 Bolton Grove 2 bed ground floor apt  82.3 sq. m Reserved\r\n2 Bolton Grove 2 bed duplex  91.7 sq. m. Reserved\r\n3 Bolton Grove 3 bed duplex  119.4 sq. m €585,000\r\n4 Bolton Grove 2 bed ground floor apt  84.6 sq. m. €500,000\r\n5 Bolton Grove 3 bed duplex  115.3 sq. m €570,000\r\n6 Bolton Grove 2 bed ground floor apt  84.6 sq. m. €500,000\r\n\r\n\r\n3 x 2-bedroom Ground floor units each with individual south facing gardens\r\n\r\n2 x 3-bedroom duplex units with private balcony\r\n\r\n1 x 2-bedroom duplex unit with private balcony\r\n\r\nQuality retail facilities nearby including Dundrum Shopping Centre, Rathfarnham Shopping\r\nCentre, RathfarnhamVillage and Nutgrove Retail Park\r\n\r\nPremier location close to St. Ends Park, Marlay Park, Bushy Park and the beautiful Dublin Mountains\r\n\r\nSpecification and Features:\r\n\r\nBER ‘A2’ rated energy efficient homes\r\n\r\n10 Year structural Homebond guarantee\r\n\r\nMaintenance free brick & render exterior\r\n\r\nRationel ‘Auraplus’ Aluclad windows\r\n\r\nAll windows and external doors are wired\r\nfor an intruder alarm\r\n\r\nCobble lock parking\r\n\r\nHigh standard of sound and heat insulation\r\n\r\nInnovative A-Class air-to-water heating system, providing energy-efficient central heating\r\nand hot water, at proven reduced energy cost\r\n\r\nGenerous electrical specification with recess LED lights and additionally television points\r\nin master bedroom\r\n\r\nWired for high-speed fibre broadband\r\n\r\nChoice of superb fitted kitchen colour with will include Quartz worktop and up stand\r\nand concealed under unit lighting\r\n\r\nGenerous kitchen appliance package with integrated appliances\r\n\r\nLuxurious built-in wardrobes to all bedrooms\r\n\r\nStylish bathrooms and En Suites with contemporary bathroom suites and generous tiling\r\n\r\nHeated towel rails standard to main bathroom and En Suite",
            "bathrooms": 0,
            "letting": null,
            "roomDetailsApproved": false,
            "summary": "• BER ‘A2’ rated energy efficient homes\r• 10 Year structural Homebond guarantee\r• Maintenance free brick & render exterior\r• Rationel ‘Auraplus’ Aluclad windows\r• All windows and external doors are wired for an intruder alarm\r• Cobble lock parking\r• High standard of sound and heat insulation\r• Innovative A-Class air-to-water heating system, providing energy-efficient central heating and hot water, at proven reduced energy cost\r• Generous electrical specification with recess LED lights and additionally television points in master bedroom\r• Wired for high-speed fibre broadband\r• Choice of superb fitted kitchen colour with will include Quartz worktop and up stand and concealed under unit lighting\r• Generous kitchen appliance package with integrated appliances\r• Luxurious built-in wardrobes to all bedrooms\r• Stylish bathrooms and En Suites with contemporary bathroom suites and generous tiling\r• Heated towel rails standard to main bathroom and En Suite",
            "lostInstructionDate": null,
            "fromArchive": false,
            "commercial": null,
            "externalArea": null,
            "address": {
                "buildingName": "",
                "line4": "Dublin 16",
                "postcode": "",
                "buildingNumber": "1-6",
                "line3": "Rathfarnham",
                "line2": "Ballyboden Road",
                "line1": "Bolton Grove",
                "countryId": "IE",
                "geolocation": {
                    "latitude": 53.284909,
                    "longitude": -6.295156
                }
            },
            "areaId": "DBL",
            "councilTax": null,
            "rooms": [],
            "boardStatus": null,
            "description": "Introducing Bolton Grove, an exclusive development of 6 brand new luxury homes smartly nestled off the Ballyboden Road next to the exclusive Bolton Hall development. Bolton Grove provides a choice of homes ranging in size from 82 sq. m. to 119 sq. m. These new A rated homes have been architecturally designed to a very high specification and come with many superb features....\r\n\r\nSituated just off Ballyboden Road in the heart of Rathfarnham, Bolton Grove is within close proximity to some wonderful local amenities including St. Endas Park, Rathfarnham Village and Rathfarnham Shopping Centre to name but a few. Neighbourhood shops are available nearby on Taylors Lane while the exclusive villages of Templeogue, Terenure and Rathgar are also withing easy reach. Local primary and secondary schools are in plentiful supply and Dublin Bus provide excellent public transport facilities to the city centre and surrounding areas. The M50 motorway is also very accessible allowing easy access to the nation’s national road network. \r\n\r\n\r\n3 x 2-bedroom Ground floor units each with individual south facing gardens\r\n 2 x 3-bedroom duplex units with private balcony\r1 x 2-bedroom duplex unit with private balcony\r\r\nQuality retail facilities nearby including Dundrum Shopping Centre, Rathfarnham Shopping Centre, Rathfarnham Village and Nutgrove Retail Park\r\r\nPremier location close to St. Endas Park, Marlay Park, Bushy Park and the beautiful Dublin Mountains",
            "regional": null,
            "localAuthorityCompanyId": null,
            "videoCaption": null,
            "officeIds": [
                "RED"
            ],
            "isExternal": false,
            "boardUpdated": null,
            "strapline": null,
            "url": "https://eddempsey.ie/properties/1-6-bolton-grove-ballyboden-road-rathfarnham-dublin-16/",
            "video2Caption": null,
            "serviceCharge": 0,
            "serviceChargeComment": null,
            "video2Url": null,
            "parkingSpaces": 0,
            "id": "RED210630",
            "internalArea": null,
            "modified": "2022-05-17T16:41:11Z",
            "age": [
                "new"
            ],
            "alternateId": null,
            "specialFeatures": [],
            "selling": {
                "agency": "soleAgent",
                "recommendedPrice": 0,
                "exchangedCompanyFee": null,
                "reservationFee": 0,
                "accountPaid": null,
                "fee": null,
                "vendorId": "RED210630",
                "agencyId": "SA",
                "completed": null,
                "disposal": "privateTreaty",
                "instructed": "2021-10-26",
                "exchangedOfficeId": "",
                "price": 0,
                "qualifier": "askingPrice",
                "exchanged": null,
                "exchangedPrice": 0,
                "priceTo": 0,
                "decoration": [
                    "veryGood"
                ],
                "tenure": null,
                "status": "valuation",
                "brochureId": null
            },
            "groundRentComment": null,
            "created": "2021-08-26T07:45:38Z",
            "lostInstructionNote": null,
            "_eTag": "\"BB149E5833693F2132618F454C1A30A6\"",
            "style": [
                "groundFloor",
                "upperFloor"
            ],
            "receptions": 0,
            "locality": [],
            "bedrooms": 2,
            "situation": [],
            "marketingMode": "selling",
            "epc": {
                "exempt": false,
                "eer": 0,
                "eirPotential": 0,
                "eerPotential": 0,
                "eir": 0
            },
            "negotiatorId": "BMC",
            "type": [
                "flatApartment"
            ],
            "featuredImageUrl": "https://assets.reapit.net/edd/live/pictures/RED/21/RED210630_14.jpg"
        },
        "diff": {
            "modified": [
                "2022-05-17T16:41:11Z",
                "2022-06-08T17:27:15Z"
            ],
            "selling": {
                "status": [
                    "valuation",
                    "forSale"
                ],
                "publicBrochureUrl": [
                    null
                ],
                "sharedOwnership": [
                    {
                        "sharedPercentage": null,
                        "rent": null,
                        "rentFrequency": ""
                    }
                ]
            },
            "_eTag": [
                "\"BB149E5833693F2132618F454C1A30A6\"",
                "\"5D41CF1B4696A9BE054E47BEF4BBB745\""
            ],
            "bedroomsMax": [
                3
            ],
            "groundRentReviewDate": [
                null
            ],
            "groundRentIncrease": [
                null
            ]
        },
        "SendAttempts": 1
    }
}

**Comment by plittlewood-rpt** [comment]
We should review this alongside #5671 with a view to possibly starting to include links AND extrasField data in our indexed copy of customer data. Doing both at the same time will mean having to reindex the whole database only once, so we should consider these items linked. moving to Not Ready on that basis so we can scope of the full requirements as one project

**Comment by colearymyhome** [comment]
To elaborate further,

This request is to add the subplots field that I'm told exists in the API response to the Webhook.

There doesn't seem to be a corresponding link in the subPlot's API response or webhook that identifies the Development

**Comment by plittlewood-rpt** [comment]
Hi @colearymyhome there are two ways we can deal with this but it depends what your intention is when a notification comes through.

- We can look at adding the `_links` collection to the webhook payload. We deliberately took these out as they are quite bloatey
- We can add a new property to the PropertyModel as per the preliminary specification above. This will be included in the webhook notifications and allow you to understand the type of property. 

What is your intended action on receipt of a webhook that fits this scenario? If you will want to refetch all properties in a development or access the site master record, it's likely option 2 won't actually won't for you are wanting to do.

Thanks
Pete

**Comment by colearymyhome** [comment]
Hi @plittlewood-rpt 

At the moment, when a subPlot is triggered, we get to see what the master development is. However when the master webhook is triggered we don't have any oversight to any linked subPlots for that master.

**Comment by plittlewood-rpt** [comment]
Hi @colearymyhome 

Can you come back to me on the query about the action you are wanting to be able to take on receipt of a webhook for a master record or sub plot as that will feed into any potential changes we look at making here. I'm not sure what you mean by your comment above "when a subPlot is triggered, we get to see what the master development is" - can you confirm what it is that you're referring to here?

Thanks
Pete

**Comment by plittlewood-rpt** [comment]
Closed due to inactivity. Please re-open add a new issue if this is still a requirement
